public/components/Tag/TagEdit.react.js

```diff
@@ -14,23 +19,37 @@ export default class TagEdit extends React.Component {
       }));
     }
 
+    //This will contain the logic for different tag forms (keyword vs contributor etc...)
+    renderTagTypeSpecificFields() {
+
+      if (!this.props.tag.type) {
+        return false;
+      }
+
+      if (this.props.tag.type === tagTypes.topic) {
+        return (<div className="tag-edit__input-group">
+          <label className="tag-edit__input-group__header">Section</label>
+          <SectionSelect selectedId={this.props.tag.section} sections={this.props.sections} onChange={this.onUpdateSection.bind(this)}/>
+        </div>);
+      }
+    }
```

Comment:
Technically section is not a topic specific field, but happy to leave it here now as an example of different fields for types (and given we've only got topics currently)